nalgebra-sparse/src/pattern.rs

```diff
@@ -153,7 +153,9 @@ impl SparsityPattern {
                     return Err(NonmonotonicOffsets);
                 }
 
-                let minor_indices = &minor_indices[range_start..range_end];
+                let minor_indices = minor_indices
+                    .get(range_start..range_end)
+                    .ok_or(MajorIndexOutOfBounds)?;
```

Comment:
Is this a blocker for the serialization stuff? Because this will definitely collide with #1015...